strapi/strapi
# Add mysql2 to available knex drivers

### What does it do?

- Attempt to use `mysql2` db driver if it is available
- Adds github workflow to run API tests with `mysql2` in addition to `mysql`
- Adds sqlite-legacy as options in tests commands

Note: This PR keeps `mysql` as the default for new Strapi projects, but because it adds `mysql2` to the getstarted project, the getstarted project will be using `mysql2`. It was necessary to add `mysql2` to some package within the monorepo in order for the API tests to run properly -- Knex loads drivers from the monorepo root node_modules rather than the testApp/node_modules and getstarted seemed the best place to put it.

However, we should consider moving to `mysql2` as the default for new Strapi projects similar to what was done for `better-sqlite3`.

### Why is it needed?

`mysql2` has been added because it is necessary for mysql 8 auth, as well as performance improvements and additional features.

### How to test it?

`yarn add mysql2` in a MySQL Strapi project and `mysql2` will be used as the db driver. (Alternatively, `MYSQL_PKG=mysql2` env variable can be set to force `mysql2` to be used over `mysql`)

The sqlite packages should continue to work as before.

### Related issue(s)/PR(s)

Fixes https://github.com/strapi/strapi/issues/13774

Initial PR fix from community user: https://github.com/strapi/strapi/pull/13775
(Missing update to tests, but credit to @benderillo for getting the ball rolling)

[Documentation PR](https://github.com/strapi/documentation/pull/1592)
